shared_model/CMakeLists.txt

```diff
@@ -12,6 +12,34 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+cmake_minimum_required(VERSION 3.5.1)
+
+PROJECT(shared_model C CXX)
+
+SET(IROHA_SCHEMA_DIR "${PROJECT_SOURCE_DIR}/../schema")
+include_directories(
+        ${PROJECT_SOURCE_DIR}/../libs
+        ${IROHA_SCHEMA_DIR}
+        ${PROJECT_SOURCE_DIR}
+)
+
+if (NOT IROHA_ROOT_PROJECT)
+    SET(CMAKE_POSITION_INDEPENDENT_CODE TRUE)
+    SET(CMAKE_CXX_FLAGS "-std=c++1y -Wall")
+    SET(CMAKE_CXX_FLAGS_RELEASE "-O3")
+    SET(CMAKE_CXX_FLAGS_DEBUG   "-g -Wextra -Wno-unused-parameter -Wno-deprecated-declarations -O0")
+    SET(CMAKE_CXX_OUTPUT_EXTENSION_REPLACE 1)
+
+    SET(CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake/Modules)
+
+    add_definitions(-DDISABLE_BACKWARD)
+
+    include(FeatureSummary)
+    include(cmake/functions.cmake)
+    include(cmake/dependencies.cmake)
+        
+    add_subdirectory(../schema build)
```

Comment:
Because `../libs` is not added, `generator` library is not built, which will probably result in unresolved symbols. It seems that testing this behaviour will require moving shared library tests to shared_library directory, adding gtest, and TESTING flag.

Reply:
Also `add_subdirectory(../libs libs)` seems necessary.